### PR TITLE
fix: navbar orientation on small screen sizes

### PR DESCRIPTION
**Pull request by hermannm**
## Proposed Changes

- refactor Navbar component to no longer show links in reverse order on small screen sizes
  - also refactor various code smells
- remove redundant 'home' link (logo leads home)

Previously: | Now:
--- | ---
image | image

## Types of Changes

- [ ] New feature
- [x] Bug fix
- [ ] Enhancement/optimization
- [ ] Refactor
- [x] Style
- [ ] Build/dependencies
- [ ] Other

## Checklist

- [ ] I have added tests to cover my changes (for features/bugs)
- [x] I am pleased with the readability of the code (if not, state where you need input)
- [x] I have tested the changes and verified that they work and do not break anything (to the best of my ability)

### CODE REVIEW

File: frontend/src/components/navbar/Navbar.tsx [hunk 1]
```diff
@@ -87,7 +87,15 @@ const Navbar: React.FC = () => {
   const classes = useStyles();
   const router = useRouter();
 
-  const [openDrawer, setOpenDrawer] = React.useState(false);
+  const [openDrawer, setOpenDrawer] = useState(false);
+
+  const { error, loading, data: userData } = useQuery<{ user: User }>(GET_USER);
```
Comment (larwaa) [inline frontend/src/components/navbar/Navbar.tsx]:
```suggestion
  const { error, loading, data } = useQuery<{ user: User | null }>(GET_USER);
```

File: frontend/src/components/navbar/Navbar.tsx [hunk 2]
```diff
@@ -87,7 +87,15 @@ const Navbar: React.FC = () => {
...
+
+  const [loggedIn, setLoggedIn] = useState<boolean>(false);
+
+  useEffect(() => {
+    setLoggedIn(!error && !loading && userData?.user?.firstName !== undefined);
+  }, [error, loading, userData]);
```
Comment (larwaa) [inline frontend/src/components/navbar/Navbar.tsx]:
```suggestion
  const loggedIn = // something based on if user.firstName exists or not
```

File: frontend/src/components/navbar/Navbar.tsx [hunk 3]
```diff
@@ -109,7 +112,8 @@ const Navbar: React.FC = () => {
                 </Box>
               </Box>
               <div className={classes.sectionDesktop}>
-                <NavbarLinks></NavbarLinks>
+                <NavbarLinks loggedIn={loggedIn} />
+                <NavbarUser loggedIn={loggedIn} username={loggedIn ? data?.user?.firstName ?? "" : undefined} />
```
Comment (larwaa) [inline frontend/src/components/navbar/Navbar.tsx:116]:
```suggestion
                <NavbarUser loggedIn={loggedIn} username={data?.user?.firstName} />
```

File: frontend/src/components/navbar/Navbar.tsx [hunk 4]
```diff
@@ -127,7 +131,8 @@ const Navbar: React.FC = () => {
         open={openDrawer}
         onClose={() => setOpenDrawer(false)}
       >
-        <NavbarLinks />
+        <NavbarUser loggedIn={loggedIn} username={loggedIn ? data?.user?.firstName ?? "" : undefined} />
```
Comment (larwaa) [inline frontend/src/components/navbar/Navbar.tsx:134]:
```suggestion
        <NavbarUser loggedIn={loggedIn} username={data?.user?.firstName} />
```

**Review by larwaa — CHANGES_REQUESTED** [review]

File: frontend/src/components/navbar/Navbar.tsx (hunk 1 above)
Comment (hermannm) [reply frontend/src/components/navbar/Navbar.tsx]: Fixed in 3f589b5.

File: frontend/src/components/navbar/Navbar.tsx (hunk 2 above)
Comment (hermannm) [reply frontend/src/components/navbar/Navbar.tsx]: Done.

File: frontend/src/components/navbar/Navbar.tsx (hunk 3 above)
Comment (hermannm) [reply frontend/src/components/navbar/Navbar.tsx:116]: `firstName` can be `undefined`.

**Review by larwaa — CHANGES_REQUESTED** [review]

File: frontend/src/components/navbar/NavbarLinks/NavbarUser.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        <>
+          <Button
+            className={[classes.navItem, classes.user].join(" ")}
+            aria-label="account of current user"
+            aria-controls="menu-appbar"
+            aria-haspopup="true"
+            onClick={handleMenu}
+            startIcon={<AccountCircleOutlined fontSize="small" />}
+          >
+            {loggedIn ? username : ""}
```
[the remark this answers is not in the document]
Comment (hermannm) [reply frontend/src/components/navbar/NavbarLinks/NavbarUser.tsx]: Well spotted! Changed in 0c009f3.

File: frontend/src/components/navbar/Navbar.tsx (hunk 3 above)
Comment (hermannm) [reply frontend/src/components/navbar/Navbar.tsx]: Nevermind, you're right! Changed in 91a3ba2.

**Review by larwaa — APPROVED** [review]

**Review by larwaa — APPROVED** [review]
LGTM

**Pull request merged**